src/main/java/net/imagej/ops/image/integral/AbstractIntegralImg.java

```diff
@@ -90,29 +90,47 @@ public void compute1(RandomAccessibleInterval<I> input,
 					integralAdd, i);
 			}
 		}
-		
+
 		RandomAccessibleInterval<? extends RealType<?>> generalizedInput = input; // FIXME
-		
+
 		// Create integral image
 		for (int i=0; i < input.numDimensions(); ++i) {
 			// Slicewise integral addition in one direction
 			slicewiseOps[i].compute1(generalizedInput, output);
-			generalizedInput = output;
+			generalizedInput = ops().copy().rai(output);
```

Comment:
can we initialize the `copy` first? Also: Do we really need to copy? can't we just do it during the first iteration?